feat: add configuration file support for API key management

lhr000lhr opened the pull request:
- Add config.json and config.template.json for secure API key storage
- Update DefaultEnvInitializer to load API keys from JSON config file
- Add Config struct for JSON decoding with proper error handling
- Include fallback mechanism when config file is missing
- Update .gitignore to prevent accidental commit of sensitive config
- Update README with clear setup instructions for both config file and direct code approaches
- Improve security by separating API keys from source code

BREAKING CHANGE: API key is now loaded from config.json instead of hardcoded value

lhr000lhr committed:
Add configuration file support and update README for API key management

- Introduced `config.json` for API key storage, with a template file `config.template.json`.
- Updated `DefaultEnvInitializer.swift` to load API keys from the configuration file.
- Enhanced README with instructions for using the configuration file and emphasized the importance of not committing sensitive information.
- Added background mode support in `Info.plist` for audio processing.
- Updated `.gitignore` to include `config.json` to prevent accidental commits.

chen-argmax commented on Playground.xcodeproj/project.pbxproj:
```diff
@@ -465,6 +469,7 @@
 				MACOSX_DEPLOYMENT_TARGET = 14.2;
 				MARKETING_VERSION = 0.0.1;
 				PRODUCT_BUNDLE_IDENTIFIER = "com.argmaxinc.example.Playground${DEVELOPMENT_TEAM}";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=iphoneos*]" = com.argmaxinc.example.Playground;
```
is this required?

lhr000lhr replied:
Yeah, Xcode generated this part

chen-argmax commented on Playground.xcodeproj/project.pbxproj:
```diff
@@ -490,7 +495,7 @@
 				DEPLOYMENT_POSTPROCESSING = NO;
 				DEVELOPMENT_ASSET_PATHS = "\"Playground/Preview Content\"";
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = 2A7SXV3WJH;
```
please remove your development team :)

chen-argmax commented on Playground.xcodeproj/project.pbxproj:
```diff
@@ -517,6 +522,7 @@
 				MACOSX_DEPLOYMENT_TARGET = 14.2;
 				MARKETING_VERSION = 0.0.1;
 				PRODUCT_BUNDLE_IDENTIFIER = "com.argmaxinc.example.Playground${DEVELOPMENT_TEAM}";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=iphoneos*]" = com.argmaxinc.example.Playground;
```
ditto

chen-argmax commented on README.md:
````diff
@@ -33,17 +33,28 @@ In Xcode, select your app target and go to **Signing & Capabilities**. Choose yo
+   ```json
+   {
+       "apiKey": "your_actual_api_key_here",
+       "huggingFaceToken": null
````
we just open sourced our huggingface repo, so huggingfacetoken is no longer needded, we can remove this field

lhr000lhr replied:
yea I also removed huggingFaceToken form the PlainTextAPIKeyProvider now

chen-argmax reviewed (changes requested):
thanks for the change! lgtm just a couple of fields remove

lhr000lhr committed:
Refactor API key management by removing huggingFaceToken from configuration

- Updated `README.md` to reflect the removal of `huggingFaceToken` from the API key configuration.
- Modified `DefaultEnvInitializer.swift` to eliminate references to `huggingFaceToken` when initializing the API key provider.
- Adjusted `config.template.json` to remove `huggingFaceToken` from the template structure.
- Cleared `DEVELOPMENT_TEAM` entries in `project.pbxproj` for better configuration management.

chen-argmax reviewed (approved):
thanks for the contribution!
